File: go/cadenas/cadenas.go
package main

import "fmt"

/* función que regresa una cadena con un saludo */
func hola() string{
  return "Hola"
}

/*
  param word : string

  función que te dice si la cadena word es palindromo
  regresa true si es palindromo
  regresa false en otro caso
*/
func palindromo(word string) bool{
  f := 0
  l := len(word)-1
  for f < l {
    if word[f] != word[l] {
      return false
    }
    f+=1
    l-=1
  }
  return true
}

/*
  param word : string

  función que invierte la cadena word
  regresa la cadena invertida
*/
func invertir(word string) string{
  i := 0
  l := len(word)
  f := l-1
  inv_str := ""
  for i < l {
    inv_str += string(word[f])
    i += 1
    f -= 1
  }
  return inv_str
}

func main() {
  fmt.Println(hola())
  
  fmt.Print("Escriba una palabra   ")
  var word string
  fmt.Scanf("%s", &word)
  fmt.Println("Es palindromo :  ", palindromo(word))
  inv_str := invertir(word)
  fmt.Println("cadena invertida  ", inv_str)
  
}

File: clojure/cadenas/cadenas.clj
;; Función que regresa un saludo
(defn hola []
  "Hola")

;; Funcion que verifica si una palabra es palindromo
;; @param x palabra a evaluar
;; @returns true si lo es, false en otro caso
(defn palindromo [x]
  (= (seq x) (reverse x)))

;; Funcion que invierte una cadena
;; @param x palabra a evaluar
;; @returns cadena invertida
(defn reversa
  ([] (str))
  ([x] (reverse x)))

;; Función que cuenta el número de repeticiones de una carácter en una palabra
;; @param word  palabra a evaluar
;; @param ch    carácter a buscar
;; @param c     contador de repeticiones
;; @returns el contador con el número de repeticiones
(defn repet [word ch c]
  (cond
    (= 0 (count word)) c
    :else (cond
    	    (= ch (str (first word))) (repet (rest word) ch (+ 1 c))
	    :else (repet (rest word) ch c))))

;; Función que llama a contar el número de repeticiones
;; del carpacter y en la palabra x
(defn repeticiones [x y]
  (repet x y 0))

(println (hola))
(println "Escribe una palabra")
(let [name (read-line)]
  (println "Es palindromo :" (palindromo name))
  (println "Palabra invertida" (reversa name))
  (println "Carácter a buscar")
  (def ch (read-line))
  (println "repeticiones de" ch ":" (repeticiones name ch)))